src/applications/check-in/components/tests/ActionItemDisplay.unit.spec.jsx

```diff
@@ -0,0 +1,109 @@
+import React from 'react';
+import { expect } from 'chai';
+import { render } from '@testing-library/react';
+import sinon from 'sinon';
+import CheckInProvider from '../../tests/unit/utils/CheckInProvider';
+import {
+  multipleAppointments,
+  singleAppointment,
+} from '../../tests/unit/mocks/mock-appointments';
+import ActionItemDisplay from '../ActionItemDisplay';
+import * as appointmentModule from '../../utils/appointment';
+
+describe('unified check-in experience', () => {
+  describe('ActionItemDisplay', () => {
+    it('displays the what to do next component if the app is day of', () => {
+      const mockstore = {
+        upcomingAppointments: multipleAppointments,
+        app: 'dayOf',
+      };
+      const { getByTestId } = render(
+        <CheckInProvider store={mockstore}>
+          <ActionItemDisplay />
+        </CheckInProvider>,
+      );
+      expect(getByTestId('what-to-do-next')).to.exist;
+    });
+    it('displays the what to do next component if the app is day of and pre check in is complete', () => {
+      const mockstore = {
+        upcomingAppointments: singleAppointment,
+        app: 'dayOf',
+      };
+      const preCheckinAlreadyCompletedStub = sinon
+        .stub(appointmentModule, 'preCheckinAlreadyCompleted')
+        .returns(true);
+      const { getByTestId } = render(
+        <CheckInProvider store={mockstore}>
+          <ActionItemDisplay />
+        </CheckInProvider>,
+      );
+
+      expect(getByTestId('what-to-do-next')).to.exist;
+
+      preCheckinAlreadyCompletedStub.restore();
+    });
+    it('displays the pre-check-in-success-alert if there demographics are up to date', () => {
+      const mockstore = {
+        upcomingAppointments: singleAppointment,
+        app: 'preCheckIn',
+      };
+      const preCheckinAlreadyCompletedStub = sinon
+        .stub(appointmentModule, 'preCheckinAlreadyCompleted')
+        .returns(true);
+      const { getByTestId } = render(
+        <CheckInProvider store={mockstore}>
+          <ActionItemDisplay />
+        </CheckInProvider>,
+      );
+
+      expect(getByTestId('pre-check-in-success-alert')).to.exist;
```

Comment:
Oh I see what I was looking for in the last test. I feel like these should just be combined

Reply:
This one is testing that the success alert shows up on the pre-check-in app if demos are up to date.